src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs

```diff
@@ -1112,6 +1112,233 @@ private static StringBuilder FormatStringBuilder(DateTime dateTime, ReadOnlySpan
             return FormatCustomized(dateTime, format, dtfi, offset, result: null);
         }
 
+        internal static bool IsValidCustomDateFormat(ReadOnlySpan<char> format, bool allowThrow)
+        {
+            int length = format.Length;
+            int i = 0;
+
+            while (i < length)
+            {
+                switch (format[i])
+                {
+                    case '\\':
+                        if (i == length - 1)
+                        {
+                            if (allowThrow)
+                            {
+                                throw new FormatException(SR.Format_InvalidString);
+                            }
+
+                            return false;
+                        }
+                        i += 2;
```

Comment:
why `i += 2`? if we had `"\:"` as the format this wouldn't catch at as invalid, right?

Reply:
`\` is escaping the next character. it will show up in the formatted string a literal.

Reply:
Ah ok... makes sense.

Reply:
No. the `\` would be just handy to escape the known formatting characters (e.g. `h`, `m`,...etc.) but nothing prevent you escaping any other characters.

Reply:
Ok sounds good. Thanks for elaborating.